File: OsmAnd/src/net/osmand/plus/track/Gpx3DWallColorType.java
package net.osmand.plus.track;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.annotation.StringRes;

import net.osmand.plus.R;

public enum Gpx3DWallColorType {

	NONE("none", R.string.shared_string_none),
	SOLID("solid", R.string.track_coloring_solid),
	DOWNWARD_GRADIENT("downward_gradient", R.string.downward_gradient),
	UPWARD_GRADIENT("upward_gradient", R.string.upward_gradient),
	ALTITUDE("altitude", R.string.altitude),
	SLOPE("slope", R.string.shared_string_slope),
	SPEED("speed", R.string.shared_string_speed);

	private final String typeName;
	private final int displayNameResId;

	Gpx3DWallColorType(@NonNull String typeName, @StringRes int displayNameResId) {
		this.typeName = typeName;
		this.displayNameResId = displayNameResId;
	}

	@NonNull
	public static Gpx3DWallColorType get3DWallColorType(@Nullable String typeName) {
		for (Gpx3DWallColorType type : values()) {
			if (type.typeName.equalsIgnoreCase(typeName)) {
				return type;
			}
		}
		return NONE;
	}

	@NonNull
	public String getTypeName() {
		return typeName;
	}

	@StringRes
	public int getDisplayNameResId() {
		return displayNameResId;
	}

	public boolean isGradient() {
		return this == ALTITUDE || this == SLOPE || this == SPEED;
	}

	public boolean isVerticalGradient() {
		return this == UPWARD_GRADIENT || this == DOWNWARD_GRADIENT;
	}
}

File: OsmAnd-shared/src/commonMain/kotlin/net/osmand/shared/routing/Gpx3DWallColorType.kt
package net.osmand.shared.routing

enum class Gpx3DWallColorType(val typeName: String, val displayNameResId: String) {

	NONE("none", "shared_string_none"),
	SOLID("solid", "track_coloring_solid"),
	DOWNWARD_GRADIENT("downward_gradient", "downward_gradient"),
	UPWARD_GRADIENT("upward_gradient", "upward_gradient"),
	ALTITUDE("altitude", "altitude"),
	SLOPE("slope", "shared_string_slope"),
	SPEED("speed", "shared_string_speed");


	companion object {
		fun get3DWallColorType(typeName: String?): Gpx3DWallColorType {
			for (type in Gpx3DWallColorType.entries) {
				if (type.typeName.equals(typeName, ignoreCase = true)) {
					return type
				}
			}
			return NONE
		}
	}

	open fun isGradient(): Boolean {
		return this == ALTITUDE || this == SLOPE || this == SPEED
	}

	open fun isVerticalGradient(): Boolean {
		return this == UPWARD_GRADIENT || this == DOWNWARD_GRADIENT
	}

}